### PR TITLE
Fixed restore validation

### PR DESCRIPTION
**Pull request by ak-px**
**What this PR does / why we need it**:
Removed go routine for concurrent restore , as createRestoreValidation method will fail if concurrent call are made.

`
2404 | Mar 2, 2024, 12:03:45 PM | failed to switch to context. RefreshDriverEndpoints Error: [failed to update node [ip-10-13-155-101.pwx.purestorage.com], Err: node to be updated does not exist] |   |   |  
-- | -- | -- | -- | -- | --
2405 | Mar 2, 2024, 12:03:45 PM | Verifying creation of restore tp-restore-tp-backup-global-aws-px-backup-system-test-b652bb51-21-hidhvu-rvbt of backup tp-backup-global-aws-px-backup-system-test-b652bb51-21-hidhvu-rvbt
`

This test was missed from validation while we bulk moved createRestore to createRestoreValidation  https://github.com/portworx/torpedo/pull/1914.  and has not been in runlist due to scale scenario.
 https://github.com/portworx/torpedo/pull/1914

**Which issue(s) this PR fixes** (optional)
Closes #PB-5890

**Special notes for your reviewer**:

Tested log:
https://jenkins.pwx.dev.purestorage.com/blue/organizations/jenkins/portworx-backup%2Fsystem-tests%2Fbyoc%2Fpx-backup-on-demand-system-test-byoc/detail/px-backup-on-demand-system-test-byoc/4729/
https://aetos.pwx.purestorage.com/resultSet/testSetID/564297

### CODE REVIEW

File: tests/backup/backup_scale_test.go [hunk 1]
```diff
@@ -134,24 +134,17 @@ var _ = Describe("{MultipleBackupLocationWithSameEndpoint}", func() {
...
 				for backupName, namespace := range userBackupMap[index] {
-					wg.Add(1)
 					restoreName := fmt.Sprintf("%s-%s", RestoreNamePrefix, backupName)
-					go createRestore(backupName, restoreName, namespace)
+					customNamespace := "custom-" + namespace + RandomString(5)
+					namespaceMapping := map[string]string{namespace: customNamespace}
+					appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+					err = CreateRestoreWithValidation(ctx, restoreName, backupName, namespaceMapping, make(map[string]string), DestinationClusterName, BackupOrgID, appContextsToBackup)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of restore %s of backup %s", restoreName, backupName))
+					restoreNames = SafeAppend(&mutex, restoreNames, restoreName).([]string)
```
Comment (snigdha-px) [inline tests/backup/backup_scale_test.go:145]: Can we do something like what is done in testcase :MultipleCustomRestoreSameTimeDiffStorageClassMapping
Take parallel restores and the validate is seperately? 
Or will it lead to testcase failure?

Comment (ak-px) [reply tests/backup/backup_scale_test.go:145]: @snigdha-px , Just checked that code, The issue here was the same context switch during validation.
Ill implement the changes to do it separately.

**Review by snigdha-px — APPROVED** [review]

File: tests/backup/backup_scale_test.go (hunk 1 above)
Comment (ak-px) [reply tests/backup/backup_scale_test.go]: Thanks for suggestion @snigdha-px .
Testing log after changes.
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/4734/console

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**